app/services/SubscriptionCacheService.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.subscription.cache.{CreateSubscriptionForDACRequest, IndividualDetails, OrganisationDetails}
+import models.subscription.{ContactInformation, ContactInformationForIndividual, ContactInformationForOrganisation, DisplaySubscriptionForDACResponse, PrimaryContact, ResponseCommon, ResponseDetail, SecondaryContact, SubscriptionForDACResponse}
+import repositories.SubscriptionCacheRepository
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SubscriptionCacheService @Inject()(cacheRepository: SubscriptionCacheRepository) {
+
+  def storeSubscriptionDetails(id: String, subscription: CreateSubscriptionForDACRequest): Future[Boolean] =
+    cacheRepository.set(id, subscription)
+
+  def retrieveSubscriptionDetails(id: String)(implicit ec: ExecutionContext): Future[Option[DisplaySubscriptionForDACResponse]] = {
+    //Fake response message from our cached details
```

Comment:
Do we need to fake a response here or is this comment redundant

Reply:
this is just a comment that we're taking out details from the cache and making it into a version of a response from another api. So basically none of the common response items will be right, we're just doing it to make this workaround fit into the flow.